lib/internal/url.js

```diff
@@ -1392,6 +1392,20 @@ function getPathFromURLPosix(url) {
   return decodeURIComponent(pathname);
 }
 
+function getPathFromURLString(path) {
+  if (typeof path === 'string') {
+    if (!path.startsWith('file://'))
+      return path;
+    path = new URL(path);
+  } else if (path == null || !path[searchParams] ||
+      !path[searchParams][searchParams]) {
+    return path;
+  }
+  if (path.protocol !== 'file:')
```

Comment:
Perhaps we can avoid this check for the first case where we already know it starts with `'file://'`?